File: algorithms/S/Smallest Value of the Rearranged Number/Smallest Value of the Rearranged Number.java
class Solution {
    public long smallestNumber(long num) {
        if(num == 0){
            return 0;
        }
        boolean isNegative = num < 0;
        num  = num < 0 ? num * -1 : num;
        
        char[] c = String.valueOf(num).toCharArray();
        Arrays.sort(c);
        String str;
        if(!isNegative){
            int non = 0;
			//if not negative we need to find out the first non-leading zero then swap with first zero
            for(; non < c.length; non++){
                if(c[non] != '0'){
                    break;
                }
            }
            char temp = c[non];
            c[non] = c[0];
            c[0] = temp;
            str = new String(c);
        }else{
            str = new String(c);
            StringBuilder sb = new StringBuilder(str);
            str = "-".concat(sb.reverse().toString());
        }
        return Long.valueOf(str);
    }
}

File: scripts/algorithms/S/Smallest Value of the Rearranged Number/Smallest Value of the Rearranged Number.cpp
// Runtime: 0 ms (Top 100.00%) | Memory: 6.1 MB (Top 63.28%)
class Solution
{
public:
    long long smallestNumber(long long num)
    {
        if (num < 0)
        {
            string s = to_string(-num);
            sort(s.rbegin(), s.rend());
            return -stoll(s);
        }
        else if (num == 0)
            return 0;
        string s = to_string(num);
        sort(s.begin(), s.end());
        int i = 0;
        while (s[i] == '0')
            i++;
        char c = s[i];
        s.erase(s.begin() + i);
        s = c + s;
        return stoll(s);
    }
};